pandas/core/indexes/base.py

```diff
@@ -4040,15 +4040,17 @@ def _coerce_to_ndarray(cls, data):
             When the data passed in is a scalar.
         """
 
-        if not isinstance(data, (np.ndarray, Index)):
-            if data is None or is_scalar(data):
-                raise cls._scalar_data_error(data)
+        if isinstance(data, (np.ndarray, Index)):
```

Comment:
can you add types for dtype Optional[np.dtype] here and update the doc-string with the new arg (also need to make a Parameters / Returns section as well)